audio/play.go

```diff
@@ -22,94 +21,159 @@ var PlaybackStartTimeout = 1 * time.Second
+					if e.Playback.ID != id {
+						Logger.Debug("Ignoring unrelated playback")
+						continue PlaybackStartLoop
+					}
+					Logger.Debug("Playback started")
+					break PlaybackStartLoop
+				case "PlaybackFinished":
+					e := v.(*v2.PlaybackFinished)
+					if e.Playback.ID != id {
+						Logger.Debug("Ignoring unrelated playback")
+						continue PlaybackStartLoop
+					}
+					Logger.Debug("Playback stopped (before PlaybackStated received)")
+					close(pb.startCh)
+					close(pb.stopCh)
+					return
+				default:
+					Logger.Debug("Unhandled e.Type", v.GetType())
 					continue PlaybackStartLoop
 				}
-				Logger.Debug("Playback stopped (before PlaybackStated received)")
-				return nil
-			default:
-				Logger.Debug("Unhandled e.Type", v.GetType())
-				continue PlaybackStartLoop
+			case <-startTimer:
+				Logger.Error("Playback timed out")
+				pb.err = timeoutErr{"Timeout waiting for start of playback"}
+				close(pb.startCh)
+				close(pb.stopCh)
+				return
 			}
-		case <-startTimer:
-			Logger.Error("Playback timed out")
-			return fmt.Errorf("Timeout waiting for start of playback")
 		}
-	}
 
-	// Playback has started.  Wait for it to finish
-	stopTimer := time.After(MaxPlaybackTime)
-PlaybackStopLoop:
-	for {
-		select {
-		case <-ctx.Done():
-			return nil
-		case v := <-s.C:
-			if v == nil {
-				Logger.Debug("Nil event received")
-				continue PlaybackStopLoop
-			}
-			switch v.GetType() {
-			case "PlaybackFinished":
-				e := v.(*v2.PlaybackFinished)
-				if e.Playback.ID != id {
-					Logger.Debug("Ignoring unrelated playback")
+		// trigger playback start signal and defer playback stop signal
+		close(pb.startCh)
+		defer close(pb.stopCh)
+
+		// Playback has started.  Wait for it to finish
+		stopTimer := time.After(MaxPlaybackTime)
+	PlaybackStopLoop:
+		for {
+			select {
+			case <-ctx.Done():
+				pb.err = pb.ctx.Err()
+				return
+			case v := <-s.C:
+				if v == nil {
+					Logger.Debug("Nil event received")
+					continue PlaybackStopLoop
+				}
+				switch v.GetType() {
+				case "PlaybackFinished":
+					e := v.(*v2.PlaybackFinished)
+					if e.Playback.ID != id {
+						Logger.Debug("Ignoring unrelated playback")
+						continue PlaybackStopLoop
+					}
+					Logger.Debug("Playback stopped")
+					return
+				default:
+					Logger.Debug("Unhandled e.Type", v.GetType())
 					continue PlaybackStopLoop
 				}
-				Logger.Debug("Playback stopped")
-				return nil
-			default:
-				Logger.Debug("Unhandled e.Type", v.GetType())
-				continue PlaybackStopLoop
+			case <-stopTimer:
+				Logger.Error("Playback timed out")
+				pb.err = timeoutErr{"Timeout waiting for stop of playback"}
+				return
 			}
-		case <-stopTimer:
-			Logger.Error("Playback timed out")
-			return fmt.Errorf("Timeout waiting for stop of playback")
 		}
-	}
+	}()
+
+	return &pb, err
+}
+
+type timeoutErr struct {
+	msg string
+}
+
+func (err timeoutErr) Error() string {
+	return err.msg
+}
+
+func (err timeoutErr) IsTimeout() bool {
```

Comment:
This kind of test is better implemented as `IsTimeoutError(err error) bool`.  If you can execute the method, you already know it is a timeoutErr.